book/src/tutorial/qt-gui/qquickpainteditem.md

```diff
@@ -0,0 +1,211 @@
+# Creating a QQuickPaintedItem in Rust
+
+To render a QImage instance in QML there are two simple approaches
+
+- Subclass [`QQuickPaintedItem`](https://doc.qt.io/qt-6/qquickpainteditem.html) and use the [`QPainter`](https://doc.qt.io/qt-6/qpainter.html) interface
+- Provide a custom [`QQuickImageProvider`](https://doc.qt.io/qt-6/qquickimageprovider.html).
+
+Because `QPainter` is very versatile and easy to implement with CXX-Qt, we'll use the first approach for our use-case.
+
+✅ Add a new Rust file for the `QQuickPaintedItem` subclass (e.g. called `image_painter.rs`).
+``` console
+$ tree -I "target/"
+.
+├── build.rs
+├── Cargo.lock
+├── Cargo.toml
+├── qml
+│   └── main.qml
+└── src
+    ├── image_painter.rs
+    └── main.rs
+
+3 directories, 6 files
+```
+
+Before you do anything else, make sure to
+
+✅ Register this file with the build system
+
+Because we will be using the `#[cxx_qt::bridge]` macro, we have to register this file twice.
+
+1. In the `src/main.rs` file, add a line: `pub mod image_painter;`\
+    This registers the new file as a module of our application and makes it known to Cargo.
+2. In the `build.rs` file, add the file to the list of rust_files of the QmlModule.\
+    This registers the new file with CXX-Qt.
+    It tells CXX-Qt to generate a C++ class from it, run moc and link the resulting class into our application.
+    This is only necessary because we want to use CXX-Qt within the file.
+    Doing this second step is not necessary for a normal Rust module.
+```rust
+use cxx_qt_build::{CxxQtBuilder, QmlModule};
+
+fn main() {
+    CxxQtBuilder::new()
+        // - Qt Core is always linked
+        // - Qt Qml is linked by enabling the qt_qml Cargo feature.
+        // - Qt Qml requires linking Qt Network on macOS
+        .qt_module("Network")
+        .qt_module("Quick")
+        .qml_module(QmlModule {
+            uri: "com.kdab.cxx_qt.demo",
+            rust_files: &["src/image_painter.rs"], // <- Add all files that
+            qml_files: &["qml/main.qml"],          //    include a #[cxx_qt::bridge].
+            ..Default::default()
+        })
+        .build();
+}
+```
+> It is a common mistake to forget to include CXX-Qt files in the `build.rs` script.
+> Make sure to try removing the file from the list so you know what happens in this case.
+>
+> ❓ What error do you get?
+
+Now we can start adding a new `QQuickPaintedItem` subclass.
+
+CXX-Qt allows us to create new `QObject` (and therefore `QQuickPaintedItem`) subclasses.
+These subclasses are defined in C++ and their data layout cannot be changed from Rust directly.
+Instead, every subclass generated by CXX-Qt contains an instance of a Rust struct.
+It is this Rust struct that therefore defines the data layout used within CXX-Qt objects.
+
+We can create a new QObject class for any Rust struct we like.
+
+✅ Create a new empty Rust struct for our class\
```

Comment:
random `\` at the end of the line? is that intentional ?

Reply:
Nope, artifact from earlier formatting. Done.